lib/mocktail/imitates_type/makes_double/declares_dry_class.rb

```diff
@@ -42,21 +44,28 @@ def define_double_methods!(dry_class, type, instance_methods)
       handles_dry_call = @handles_dry_call
       instance_methods.each do |method|
         dry_class.undef_method(method) if dry_class.method_defined?(method)
+        parameters = type.instance_method(method).parameters
+        signature = @transforms_params.transform(Call.new, params: parameters)
+        method_signature = @builds_method_signature.build(signature)
 
-        dry_class.define_method method, ->(*args, **kwargs, &block) {
-          Debug.guard_against_mocktail_accidentally_calling_mocks_if_debugging!
-          handles_dry_call.handle(Call.new(
-            singleton: false,
-            double: self,
-            original_type: type,
-            dry_type: dry_class,
-            method: method,
-            original_method: type.instance_method(method),
-            args: args,
-            kwargs: kwargs,
-            block: block
-          ))
-        }
+        # TODO: This is failing because none of self, type, dry_class, method,
+        # etc are in scope for the dry_class.
```

Comment:
It's been a while so I’m not 100% sure, but based on the TODO comment I think I jogged my memory enough to suspect that while it can be done, it can’t be done the way Mocktail currently defines methods, or even with this class_eval. If we could somehow get these to be in scope at _runtime_ for dry_class rather than here at definition time for DeclaresDryClass, it could work I think.

Also, the current error doesn't even have HandlesDryClass in scope which is more surmountable I think, by prepending Mocktail:: to it and Call.

Do you know of a way to use class_eval but to pass the binding through to share scope?

Reply:
This is moot now with `eval`, but of course `eval` has its own issues. Rubocop is yelling at me about it and while I don't think there's a security implication, the method signature is technically user input and that's being passed (interpolated) to eval.

Reply:
The beauty of making test-scoped libraries!